src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs

```diff
@@ -552,6 +550,11 @@ public void Insert(int index, Page page)
             navigationPages.Insert(index, page);
             Add(page);
             page.Navigator = this;
+            if (index == PageCount - 1)
+            {
+                var curTop = Peek();
```

Comment:
This Peek() returns the current top (which is page) not the previous top, because it is called AFTER page has been inserted.

Reply:
oops!

Reply:
Yeah, you are right.... I fixed it!